src/crypto/SecretStorage.ts

```diff
@@ -14,61 +14,97 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { EventEmitter } from 'events';
 import { logger } from '../logger';
 import * as olmlib from './olmlib';
 import { randomString } from '../randomstring';
-import { encryptAES, decryptAES } from './aes';
+import { encryptAES, decryptAES, IEncryptedPayload } from './aes';
 import { encodeBase64 } from "./olmlib";
+import { ICryptoCallbacks, MatrixClient, MatrixEvent } from '../matrix';
+import { IAddSecretStorageKeyOpts, ISecretStorageKeyInfo } from './api';
+import { EventEmitter } from 'stream';
 
-export const SECRET_STORAGE_ALGORITHM_V1_AES
-    = "m.secret_storage.v1.aes-hmac-sha2";
+export const SECRET_STORAGE_ALGORITHM_V1_AES = "m.secret_storage.v1.aes-hmac-sha2";
 
 const ZERO_STR = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
 
+export type SecretStorageKeyTuple = [keyId: string, keyInfo: ISecretStorageKeyInfo];
+
+export interface ISecretRequest {
+    requestId: string;
+    promise: Promise<string>;
+    cancel: (reason: string) => void;
+}
+
+export interface IAccountDataClient extends EventEmitter {
+    getAccountDataFromServer: (string) => Promise<any>;
+    getAccountData: (string) => object;
+    setAccountData: (string, object) => Promise<void>;
+}
+
+interface ISecretRequestInternal {
+    name: string;
+    devices: Array<string>;
```

Comment:
```suggestion
    devices: string[];
```

We seem to prefer the shorthand syntax